dll/cpl/sysdm/environment.c

```diff
@@ -45,97 +145,423 @@ GetSelectedListViewItem(HWND hwndListView)
     return -1;
 }
 
+static VOID
+AddValuesToListBox(HWND hwndDlg,
+                   PDIALOG_DATA DlgData)
+{
+    TCHAR szTemp[wcslen(DlgData->VarData->lpRawValue) + 1];
```

Comment:
Wrong. If the size of szTemp needs to be dynamic (known only at runtime), you need to use a heap allocation.

Reply:
If all compilers support this, it's fine.
It is valid after all.

Reply:
I know that recent C compilers support that, but the problem is that stack size is much more limited than heap size, and that if the string is "too long" (large size, still fine for an environment variable but too large for the remaining stack at this function call), you would get a crash, while if you do your heap allocation, you can gracefully handle the allocation failure in case there is no much memory available anymore (otherwise the heap allocation would succeed and you could continue).

Reply:
@HBelusca @learn-more MSVC complains about it anyway, I now went for heap allocation.